client-js/src/client/backend-client.js

```diff
@@ -546,11 +587,70 @@ class FirebaseBackendClient extends BackendClient {
     const id = new SubscriptionId();
     id.setValue(path);
     subscription.setId(id);
-    subscription.setTopic(topic);
+    const spineTopic = topic.internal();
+    subscription.setTopic(spineTopic);
     return subscription;
   }
 }
 
+/**
+ * A wrapper for the query sent to the server.
+ */
+class Query {
```

Comment:
@dmitrykuzmin I don't understand how come `Query` and `Topic` are both among newly added lines. Let's discuss.